game/scenes/player/player.tscn

```diff
@@ -21,7 +22,9 @@ radius = 23.0
 y_sort_enabled = true
 collision_mask = 133
 script = ExtResource("1_wyaxs")
-inventory_data = null
+item_inventory_data = null
+weapon_inventory_data = null
+equipment_inventory_data = null
```

Comment:
Maybe it's too many layers of abstraction but I wonder if it would make sense to abstract these away into an `Inventory` node that is a child of the player. And then it would deal with all of the item juggling etc.